### PR TITLE
Add ChainHeads v2 API

### PR DESCRIPTION
**Pull request by ajsutton**
## PR Description
Migrate the v1 debug chain heads API to new rest endpoint approach and add new v2 chain heads API which includes execution_optimistic.

## Fixed Issue(s)
#4968 

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/debug/GetChainHeadsV1.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+  public GetChainHeadsV1(final DataProvider dataProvider) {
+    this(dataProvider.getChainDataProvider());
+  }
+
+  public GetChainHeadsV1(final ChainDataProvider chainDataProvider) {
+    super(chainDataProvider, ROUTE, CHAIN_HEAD_TYPE_V1);
+  }
+
+  @OpenApi(
```
Comment (rolfyone) [inline data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/debug/GetChainHeadsV1.java:64]: should we be marking v1 deprecated?

File: data/serializer/src/main/java/tech/pegasys/teku/api/response/v1/debug/ChainHeadV2.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  public final UInt64 slot;
+
+  @Schema(type = "string", format = "byte", description = DESCRIPTION_BYTES32)
+  public final Bytes32 root;
+
+  @JsonProperty("execution_optimistic")
+  public final boolean execution_optimistic;
+
+  @JsonCreator
+  public ChainHeadV2(
```
Comment (rolfyone) [inline data/serializer/src/main/java/tech/pegasys/teku/api/response/v1/debug/ChainHeadV2.java]: V1 folder and V2 name? should it be moved?

File: data/serializer/src/main/java/tech/pegasys/teku/api/response/v1/debug/GetChainHeadsResponseV2.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.api.response.v1.debug;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.List;
+
+public class GetChainHeadsResponseV2 {
```
Comment (rolfyone) [inline data/serializer/src/main/java/tech/pegasys/teku/api/response/v1/debug/GetChainHeadsResponseV2.java]: would probably also move this...

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/debug/GetChainHeadsV1.java (hunk 1 above)
Comment (ajsutton) [reply data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/debug/GetChainHeadsV1.java:64]: I guess we're at the point where we're committing to this.  The PR got merged into the standard API repo so yeah...

**Review by rolfyone — APPROVED** [review]
LGTM just might want to make sure the data types are located, I'm not sure if it's quite in the right spot.

File: data/serializer/src/main/java/tech/pegasys/teku/api/response/v1/debug/ChainHeadV2.java (hunk 2 above)
Comment (ajsutton) [reply data/serializer/src/main/java/tech/pegasys/teku/api/response/v1/debug/ChainHeadV2.java]: Good point, moved all the v2 related stuff to v2 packages.

**Pull request merged**